### ISSUE TITLE
No backup password known when trying to import exported items

### ISSUE THREAD

**Issue by YoeriNijs** [freeotp/freeotp-android#321]
Just trying to import a xml export, and the import screen asks for a backup password. However, I did not provide any password somewhere, so I cannot import the items. How can I provide a password?

**Comment by justin-stephenson** [comment]
Hi, What data are you trying to import? The Import mechanism is only possible with a FreeOTP 2.X-generated backup file. On upgrade from FreeOTP 1.X to 2.X you should have been prompted to set a backup password.

**Comment by siof** [comment]
Issue creator probably don't remember password because it's needed only on backup restore - same as me. How can i change/reset backup password ? Unfortunately i can't find any help/documentation for application.

**Comment by frozencemetery** [comment]
> Hi, What data are you trying to import? The Import mechanism is only possible with a FreeOTP 2.X-generated backup file. On upgrade from FreeOTP 1.X to 2.X you should have been prompted to set a backup password.

Surely it would be better to set the backup file's password when the file is created?

**Comment by frozencemetery** [comment]
e.g., on a fresh install of freeotp 2.x, there's no prompt, so what key are backup files encrypted with?

**Comment by timjrd** [comment]
_First: thanks for this open-source dead-simple TOTP app! I didn't want to rely on Google/Microsoft Authenticator, so this is much appreciated._ 🙂

> [...] On upgrade from FreeOTP 1.X to 2.X you should have been prompted to set a backup password.

@justin-stephenson I was using FreeOTP v1 and auto-upgraded to v2. I don't remember having been prompted for a password. Now I can't create backups because upon importing them the password is unknown to me.

_Not being able to do backups is a **critical issue** for me as I can't use Google's app backups._

Thanks! 🙂

**Comment by DrKsantor** [comment]
Same problem here. Unable to import my restore file. So is there a way to restore password/restore without password?

**Comment by draker42** [comment]
Same problem for me. Just like @timjrd I have used FreeOTP for ages, and if I had been asked to set a backup password at any time I would have noted it in my password store. Cannot transfer my 2FA codes because I do not have a password to offer. Agree with @frozencemetery -- why not prompt for a password at backup? Require phone unlock code/fingerprint/whatever before exporting to secure the process. 
Also the export screen is... confusing, where is the filename? Why may I edit 'externalBackup' text? Why do arrows next to this text do nothing?
Had no issues with FreeOTP for many years, and I'm prepared to re-acquire all my 2FA codes, because that's what FreeOTP used to say (no backups). Disappointed that an apparently poorly thought-out backup procedure has been added. 3 stars.

**Comment by mihaime** [comment]
I also see this as more useful as you suggested @draker42 with prompting for a password at backup.
I ended up disabling 2FA everywhere (my problem was different...had a rooted phone where I could restore FreeOTP with Swiftbackup and no way to migrate to non-rooted phone), reinstalling my phone, then re-enabling 2FA.
A bit cumbersome but it is what it is.

**Comment by siof** [comment]
I personally switched to AuthenticatorPro (https://github.com/jamie-mh/AuthenticatorPro) by going site by site, disabling and enabling again 2FA to setup on new authenticator.

**Comment by andy-maier** [comment]
I was also not prompted for a backup password when upgrading to version 2. This issue should be fixed!

**Comment by pkocandr** [comment]
Yeah, same issue here. I'm not even sure if my app was on version 2.x, but I never entered the master password. Now I backed up the tokens, factory reset my phone, then all apps got reinstalled and I'm not able to restore the keys. At least most of the accounts have an alternative way e.g. using an SMS for the GitHub account. Otherwise I wouldn't be able to comment here now. This is really a huge fail. :-(

**Comment by walter-bd** [comment]
Yes, I did a fresh install of the app and it doesn't ask me for a password just a backup restore password which I don't have and allows me to click on cancel and start creating new 2FA without any new password

**Comment by ericprud** [comment]
I'm trying to solve this problem going forward by setting the password for future backups. I uninstalled FreeOTP and re-installing it, thinking I would be prompted to enter a password for future backups. I was indeed prompted for a backup password but it was apparently supposed to match a backup someplace on my android so I was again asked to match a password that I have no memory of setting (the phone is two months old).

Assuming one is willing to burn down all existing OTP lists and restart with fresh-generated ones, how does one set the backup password for the future?

**Comment by vvo-peytz** [comment]
> Bzzzt -- have to eat my words. I guessed at my old password and it worked -- restored all my 2FA keys, and they work (at least one does). Apologies. You get your 2 stars back -- 5 stars! I get rapped across the knuckles for failing to note I'd made a password...
> 
> I was wrong
> Agree with @frozencemetery -- why not prompt for a password at backup? Require phone unlock code/fingerprint/whatever before exporting to secure the process. Also the export screen is... confusing, where is the filename? Why may I edit 'externalBackup' text? Why do arrows next to this text do nothing?
> 
> Had no issues with FreeOTP for many years, and I'm prepared to re-acquire all my 2FA codes, because that's what FreeOTP used to say (no backups).
> 
> I was wrong

Yes, one of my passwords I'm using also have worked for me so finally I've migrated successfully.

**Comment by java4devgmail** [comment]
I am having same issue. I am switching phone aftet 5 years and I did a backup, no password asked but on restore is asking for password. I have no possibility to find original setups for each account to recreathem on the new phone, so backup or transfer somehow is the only option if it worked.

**Comment by vvo-peytz** [comment]
> so, where is the password stored? i have access to both devices and want to migrate

try all your personal passwords until you find the correct one unless you have used a random dummy pass which you didn't  save anywhere

It worked for me. Don't think the password are stored anywhere.

**Comment by ericprud** [comment]
I too had to switch to another app. AFAICT, there's no way to start over once you lose the password; the app will never again run on that device.

**Comment by milomak** [comment]
this seems to continue being a problem.

password should be set at each backup i think.

i luckily guessed but now it doesn't seem to have restored anything

**Comment by Hydranet** [comment]
I ran into this issue when switching to a new phone. I was able to make a backup but didn't know the password for the import. I would have thought the most logical thing would be to ask to create a backup password when you make the backup. I had to redo all of my 2FA enabled accounts because of this. I think I might look at another 2FA app where this isn't a problem since it's been a 1,5 years and nothing has been done with this yet.

**Comment by ueco-jb** [comment]
Why are the creators ignoring this issue?

**Comment by mihaime** [comment]
they are not, they suggest you use another software like the rest of us :)

**Comment by ueco-jb** [comment]
> i luckily guessed but now it doesn't seem to have restored anything

I second that. The restore does not work.

**Comment by GTP95** [comment]
> they are not, they suggest you use another software like the rest of us :)

Where are they suggesting that?

**Comment by milomak** [comment]
i think it's other users that have suggested using something different.

that's what i certainly did. i was using bitwarden as my password manager
so i think i went with theirs
